lib/commands/command-update-nodebuild.bash

```diff
@@ -0,0 +1,31 @@
+#! /usr/bin/env bash
+
+set -eu -o pipefail
+
+source "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/../utils.sh"
+
+: "${ASDF_NODEJS_NODEBUILD_HOME=$ASDF_NODEJS_PLUGIN_DIR/.node-build}"
+: "${ASDF_NODEJS_NODEBUILD_REPOSITORY=https://github.com/nodenv/node-build.git}"
+
+ensure_updated_project() {
+  local pull_exit_code=0 output=
+
+  if ! [ -d "$ASDF_NODEJS_NODEBUILD_HOME" ]; then
+    printf "Cloning node-build...\n"
+    git clone "$ASDF_NODEJS_NODEBUILD_REPOSITORY" "$ASDF_NODEJS_NODEBUILD_HOME"
+  else
+    printf "Trying to update node-build..."
+    output=$(git -C "$ASDF_NODEJS_NODEBUILD_HOME" pull origin master 2>&1) || pull_exit_code=$?
+
+    if [ "$pull_exit_code" != 0 ]; then
+      printf "\n%s\n\n" "$output" >&2
+      printf "$(colored $RED ERROR): Pulling the node-build repository exited with code %s\n" "$pull_exit_code" >&2
+      printf "Please check if the git repository at %s doesn't have any changes or anything\nthat might not allow a git pull\n" "$ASDF_NODEJS_NODEBUILD_HOME" >&2
+      exit $pull_exit_code
+    fi
```

Comment:
What if the user wants to do a build offline from a previously downloaded nodejs version?

Reply:
Currently, the user could place the tarball of the source in `$ASDF_DATA_DIR/downloads/nodejs/<version>` and it should work. We might make this easier in the future, but `node-build` has some limitations around it

Reply:
But if the script tries to download `node-build` first and it fails, the user will not be able to install nodejs even if the archive is already present on their system.

But now I realize this plugin doesn't implement the required `bin/download` callback (https://asdf-vm.com/plugins/create.html#environment-variables). That's on me, I never got around to adding it here.

But I think it's good to support users who are offline. If a user has an archive already downloaded to `.asdf/downloads` from a past build, they should be able to install if without internet access, so I think the logic here may need to be re-worked a bit so updating node-build is skipped when the user is offline.

Reply:
Updating `node-build` is optional when installing new nodejs versions and will just pop a warning. See the function `try_to_update_nodebuild` on `bin/install`.

This file is just the custom command for calling it manually, it's used internally on `install` but we catch errors accordingly there